### PR TITLE
Apply new blobSidecar gossip rule validating index

### PR DESCRIPTION
**Pull request by tbenr**
https://github.com/ethereum/consensus-specs/pull/3525

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Review by StefanBratanov — APPROVED** [review]
LGTM. Let's wait for specs PR to be merged before merging this one?

**Comment by tbenr** [comment]
@StefanBratanov yep

**Pull request merged**
